### PR TITLE
Adding Py Dependency

### PR DESCRIPTION
**Pull request by psteyer**
Adding Py dependency see following Jira ticket for more information:
https://anaconda.atlassian.net/browse/CVE-566

## changes
- adding `py`
- updated to newer version
- removed noarch

### CODE REVIEW

**Review by ryanskeith — CHANGES_REQUESTED** [review]
I can't find a run requirement.  There is a test requirement though if you want to add it there.  I also noticed we are out of date for this package.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,6 +26,7 @@ requirements:
     - python
     - pip
     - setuptools >=40.0
+    - py
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:29]:
```suggestion
```
I would recommend to update to the latest v0.12. 
`py` is only a test dependency inside the f-strings https://github.com/search?q=repo%3Apytest-dev%2Fpytest-rerunfailures%20py.&type=code

Comment (psteyer) [reply recipe/meta.yaml]: I've updated the version and moved `py` to test.

Comment (skupr-anaconda) [reply recipe/meta.yaml]: Will it fail without `py` in `test/requires`?

**Review by ryanskeith — COMMENTED** [review]
If we are rebuilding, we should remove the noarch: python.  Also, we should update the script line to our current standard.

File: recipe/meta.yaml (hunk 1 above)
Comment (psteyer) [reply recipe/meta.yaml]: It did not fail without `py` in `test/requires`, but it does seem to require it in upstream.

From a grep I've done:
`./pytest-rerunfailures/test_pytest_rerunfailures.py: import py`

Figured better safe then sorry.

**Review by skupr-anaconda — APPROVED** [review]

**Review by ryanskeith — APPROVED** [review]
LGTM.

**Pull request merged**